govc/cluster/usage.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"text/tabwriter"
+
+	"github.com/vmware/govmomi/govc/cli"
+	"github.com/vmware/govmomi/govc/flags"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/units"
+	"github.com/vmware/govmomi/vim25/mo"
+)
+
+type usage struct {
+	*flags.DatacenterFlag
+
+	shared bool
+}
+
+func init() {
+	cli.Register("cluster.usage", &usage{})
+}
+
+func (cmd *usage) Register(ctx context.Context, f *flag.FlagSet) {
+	cmd.DatacenterFlag, ctx = flags.NewDatacenterFlag(ctx)
+	cmd.DatacenterFlag.Register(ctx, f)
+
+	f.BoolVar(&cmd.shared, "S", false, "Exclude host local storage")
+}
+
+func (cmd *usage) Usage() string {
+	return "CLUSTER"
+}
+
+func (cmd *usage) Description() string {
+	return `Cluster resource usage summary.
+
+Examples:
+  govc cluster.usage ClusterName
+  govc cluster.usage -S ClusterName # summarize shared storage only
+  govc cluster.usage -json ClusterName | jq -r .CPU.Summary.Usage`
+}
+
+func (cmd *usage) Run(ctx context.Context, f *flag.FlagSet) error {
+	finder, err := cmd.Finder()
+	if err != nil {
+		return err
+	}
+
+	obj, err := finder.ClusterComputeResource(ctx, f.Arg(0))
+	if err != nil {
+		return err
+	}
+
+	var res Usage
+	props := []string{"summary", "datastore", "host"}
```

Comment:
1. Nit: Seems like we are not using the "summary" property at all, thus could be removed.
2. I see that you are explicitly iterating over the hosts/datastores to calculate the cpu/mem/storage usage. Question: Couldn't/Shoudn't we use the GetResourceUsage method on the ClusterComputeResource object for this data?  see https://code.vmware.com/apis/358/vsphere/doc/vim.ClusterComputeResource.html#getResourceUsage (Note: Earlier I tried to correlate the data returned from this method to what we see in the UI and those did not align completely as I expected. So it is possible that what you are doing is more accurate. But I wanted to refer to this object here so that we can document at least why we did not use this for future code readers reference)

Reply:
nit fixed.  I hadn't actually seen the `GetResourceUsage` method, I can give it a try.
The cluster "summary" property was there as I was trying to use https://vdc-download.vmware.com/vmwb-repository/dcr-public/b50dcbbf-051d-4204-a3e7-e1b618c1e384/538cf2ec-b34f-4bae-a332-3820ef9e7773/vim.ComputeResource.Summary.html , but that data didn't align with the UI either.

So I opted for using the host data, which is just an aggregate of what you'd see with the govc `host.info` command.

Reply:
@sidharthsurana I'd like to get govc 0.24 released before the break, but probably won't have time to look at switching to `GetResourceUsage`. But could look at using that after the break to optimize the cluster.usage command.  What do you think?

Reply:
sure, that would work.